src/lib.rs

```diff
@@ -1138,6 +1142,54 @@ pub trait Itertools : Iterator {
         adaptors::dedup_by_with_count(self, cmp)
     }
 
+    /// Return an iterator adaptor that produces elements that appear more than once during the
+    /// iteration. Duplicates are detected using hash and equality.
+    ///
+    /// The iterator is stable, returning the duplicate items in the order in which they occur in
+    /// the adapted iterator. Each duplicate item is returned exactly once. If an item appears more
+    /// than twice, the second item is the item retained and the rest are discarded.
```

Comment:
Can you clarify the difference between this method and `unique`? Surely, it's more than *just* the `Clone` bound — otherwise we'd just remove the `Clone` bound on `unique`.

Is the difference just that `unique` produces the first encountered duplicate, and `duplicate` produces the last encountered duplicate?

Reply:
For sure, the difference is that `unique` produces a value regardless of whether it is a duplicate or not, whereas this method only produces elements that appear at least twice.

i.e `unique` on `[ 1, 2, 3 ]` produces `[1, 2, 3]` whereas `duplicate` produces no values